iothub/device/src/InternalClient.cs

```diff
@@ -819,20 +819,7 @@ public async Task SetMethodHandlerAsync(string methodName, MethodCallback method
             }
             finally
             {
-                try
-                {
-                    _methodsDictionarySemaphore.Release();
-                }
-                catch (SemaphoreFullException)
```

Comment:
This catch was part of an earlier attempt at fixing a similar bug, but it won't be needed anymore since this finally block will only be reached if the semaphore was grabbed